src/Serialization/containers.jl

```diff
@@ -25,15 +35,52 @@ function has_empty_entries(obj::T) where T <: ContainerTypes
   return false
 end
 
-function type_params(obj::S) where {T <: ContainerTypes, S <:MatVecType{T}}
+function type_params(obj::S) where {T <: ContainerTypes, S <: MatVecType{T}}
   isempty(obj) && return TypeParams(S, TypeParams(T, nothing))
 
   # empty entries can inherit params from the rest of the collection
   params = type_params.(filter(!has_empty_entries, obj))
-  @req params_all_equal(params) "Not all params of Vector or Matrix entries are the same, consider using a Tuple for serialization"
+  @req params_all_equal(params) "Not all params of Array entries are the same, consider using a Tuple for serialization"
   return TypeParams(S, params[1])
 end
 
+function save_type_params(s::SerializerState,
```

Comment:
why do we need this? shouldn't this case be already handled by https://github.com/oscar-system/Oscar.jl/blob/87141eb743d51ebee0eb03cbc7839840b92bbcdb/src/Serialization/containers.jl#L297?

Reply:
i need this so i know to store the dims of the array